src/NexusMods.App.UI/WorkspaceSystem/GridUtils.cs

```diff
@@ -85,8 +87,8 @@ internal static bool IsPerfectGrid(WorkspaceGridState gridState)
             {
                 foreach (var panelToSplit in rows)
                 {
-                    res.Add(SplitPanelInHalf(currentState, panelToSplit, splitVertically: false, inverse: false));
-                    res.Add(SplitPanelInHalf(currentState, panelToSplit, splitVertically: false, inverse: true));
+                    res1.Add(SplitPanelInHalf(currentState, panelToSplit, splitVertically: false, inverse: false));
+                    res1.Add(SplitPanelInHalf(currentState, panelToSplit, splitVertically: false, inverse: true));
```

Comment:
So I tested this and works great when you have a single panel.
If you have two panels side by side, then it offers to split the first panel on the left horizontally first, instead of offering the right most panel first, which I think would be more desirable.

I think it can be fixed by populating a list of lists in this While loop, and then inverting the order of the parent list. So the left panel split results are presented first.

Reply:
This is my test code for reference:
```csharp
        var newRowsList = new List<List<WorkspaceGridState>>();
        while (columnEnumerator.MoveNext(rowBuffer))
        {
            var column = columnEnumerator.Current;
            var rows = column.Rows;
            
            var newRows = new List<WorkspaceGridState>();
            
            // NOTE(erri120): In a horizontal layout, the rows can move independent of rows in other columns.
            if (rows.Length != maxRows)
            {
                foreach (var panelToSplit in rows)
                {
                    newRows.Add(SplitPanelInHalf(currentState, panelToSplit, splitVertically: false, inverse: false));
                    newRows.Add(SplitPanelInHalf(currentState, panelToSplit, splitVertically: false, inverse: true));
                }
            }

            // NOTE(erri120): If these conditions are met, we can split all rows of the current column in half
            // to create a new column. Note that this requires knowing the current amount of columns, so two
            // iterations are required (that's what CountColumns does).
            if (rows.Length > 1 && columnCount != maxColumns)
            {
                newRows.Add(AddColumn(currentState, column.Info, rows, inverse: false));
                newRows.Add(AddColumn(currentState, column.Info, rows, inverse: true));
            }
            newRowsList.Add(newRows);
        }

        res1.AddRange(Enumerable.Reverse(newRowsList).SelectMany(x => x));
```

Reply:
The same can be achieved using a simpler LINQ chain:

```csharp
step1Res = step1Res.Chunk(size: 2).Reverse().SelectMany(x => x).ToList();
```